test/integration/helpers_test.go

```diff
@@ -0,0 +1,168 @@
+// and patched with custom behavior to determine the health of the gateway and to return an error
+// instead of failing (at the time of its call, we don't have any test instance to make fail yet).
+func gatewayHealthCheck(ctx context.Context, client *gatewayclient.Clientset, gatewayName, namespace string) error {
+	ch := make(chan bool, 1)
+
+	timer := time.NewTimer(gatewayWaitTimeToVerifyScheduling)
+	defer timer.Stop()
+
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	for tick := ticker.C; ; {
+		select {
+		case <-timer.C:
+			return fmt.Errorf("the gateway scheduling %s/%s has never occurred", namespace, defaultGatewayName)
+		case <-tick:
+			tick = nil
+			go func() {
+				ch <- func() bool {
+					gw, err := client.GatewayV1alpha2().Gateways(namespace).Get(ctx, gatewayName, metav1.GetOptions{})
+					exitOnErr(err)
+					for _, cond := range gw.Status.Conditions {
+						if cond.Reason == string(gatewayv1alpha2.GatewayReasonReady) {
+							return true
+						}
+					}
+					return false
+				}()
+			}()
+		case v := <-ch:
+			if v {
+				return nil
+			}
+			tick = ticker.C
+		}
+	}
+}
+
+// eventuallyGatewayIsLinked checks if the specific Route (HTTP, TCP, TLS, or UDP)
+// is correctly linked to (or unlinked from) a supported gateway. In order to
+// specify whether to check linking or unlinking, the ensureLinked boolean arg must
+// be set accordingly.
+func eventuallyGatewayIsLinked(t *testing.T, c *gatewayclient.Clientset, ensureLinked bool, protocolType gatewayv1alpha2.ProtocolType, namespace, name string) {
+	var routeParents []gatewayv1alpha2.RouteParentStatus
+
+	require.Eventually(t, func() bool {
+		// gather a fresh copy of the route, given the specific protocol type
+		switch protocolType {
+		case gatewayv1alpha2.HTTPProtocolType:
+			route, err := c.GatewayV1alpha2().HTTPRoutes(namespace).Get(ctx, name, metav1.GetOptions{})
+			require.NoError(t, err)
+			routeParents = route.Status.Parents
+		case gatewayv1alpha2.TCPProtocolType:
+			route, err := c.GatewayV1alpha2().TCPRoutes(namespace).Get(ctx, name, metav1.GetOptions{})
+			require.NoError(t, err)
+			routeParents = route.Status.Parents
+		case gatewayv1alpha2.UDPProtocolType:
+			route, err := c.GatewayV1alpha2().UDPRoutes(namespace).Get(ctx, name, metav1.GetOptions{})
+			require.NoError(t, err)
+			routeParents = route.Status.Parents
+		case gatewayv1alpha2.TLSProtocolType:
+			route, err := c.GatewayV1alpha2().TLSRoutes(namespace).Get(ctx, name, metav1.GetOptions{})
+			require.NoError(t, err)
+			routeParents = route.Status.Parents
+		default:
+			t.Fatalf("protocol %s not supported", string(protocolType))
+		}
+
+		// determine if there is a link to a supported Gateway
+		for _, parentStatus := range routeParents {
+			if parentStatus.ControllerName == gateway.ControllerName {
+				// supported Gateway link was found, hence if we want to ensure
+				// the link existence return true, otherwise true
```

Comment:
> return true, otherwise true

I feel the `otherwise true` is redundant, isn't it?

```suggestion
				// supported Gateway link was found, hence if we want to ensure
				// the link existence return true
```